specification/dimensions/pricingmodel.md

```diff
@@ -0,0 +1,39 @@
+# Pricing Model
+
+Pricing Model indicates the type of unit price applied to a charge at the time it was used or purchased. Pricing Model is commonly used to differentiate between charges at on-demand vs. reduced prices when investigating optimization opportunities, like for commitment-based discount coverage.
+
+The PricingModel column MUST be present and MUST NOT be null or empty. This column is of type String and MUST be one of the allowed values. PricingModel MUST be "Commitment-Based Discount" when CommitmentDiscountId is not null. PricingModel MUST be "Dynamic" when pricing is determined by the provider and may change over time, regardless of predetermined agreement pricing.
+
+## Column ID
+
+PricingModel
+
+## Display name
+
+Pricing Model
+
+## Description
+
+Type of unit price applied to a charge at the time it was used or purchased.
+
+## Content constraints
+
+|    Constraint   |      Value       |
+|:----------------|:-----------------|
+| Column required | True             |
+| Data type       | String           |
+| Allows nulls    | False             |
+| Value format    | List of values   |
+
+Allowed values:
+
+| Value                     | Description                                                                                                                                                                                                  |
+| :------------------------ | :----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Fixed                     | Charges priced at a constant rate, regardless of quantity used or purchased.                                                                                                                                 |
+| On-Demand                 | Charges priced at the standard rate for the billing account. This includes any applicable negotiated discounts and volume/tiered pricing, but does not include dynamic or commitment-based discount pricing. |
+| Dynamic                   | Charges priced at a variable rate determined by the provider. This includes any product or service with a unit price the provider can change without notice, like interruptible or low priority resources.   |
```

Comment:
So is the key phrase here "can change without notice"? 

Because the Provider can change the cost of On-Demand as prices increase (like Azure did recently as they moved to peg it to USD) however, they gave notice for this increase.

So the difference here is that you sign up to the service without knowing the EXACT cost in advance and when it will changed?

Reply:
Is the difference that the price changes happen within a billing period here rather than across different billing periods for list price changes? Thoughts on this:
```suggestion
| Dynamic                   | Charges priced at a variable rate that is designed to change throughout the day. This includes any product or service with a unit price the provider can change at any moment without notice, like interruptible or low priority resources.   |
```

Fwiw, you do know the price you'll have. Part of creating a Spot VM (at least in Azure) is that you define a max price you're willing to pay. Not sure about other clouds.

Reply:
I find the naming Dynamic is not clear enough.

For example, I would think that GCP SUD is included in Dynamic.
I believe we had a discussion to include GCP SUD within OnDemand, and to address GCP SUD with further dimensions.

Why not just use the value "Interruptible"?

Reply:
Actually, I am not clear why Pricing Model would differentiate Spot (Dynamic/Interruptible) but not GCP SUD.

In Azure, Spot instances have no SLA: so it's not only a Pricing Model, it's also a different Product/Sku/UsageType (we are currently discussing on this naming :-)). Conversely, GCP CUD and SUD can apply to the same Product/Sku/UsageType, so it relates to a Pricing Model.

Reply:
Because SUDs are computed as a separate line item combining all discounts of all machine types for all projects, and at the end of a billing period, I'm more inclined to push it into an adjustment dimension. Something more akin to a perma-conditional credit or pricing adjustment.

Reply:
Another way to call Dynamic or spot is Spare Capacity Discount. Interesting to hear opinions about this name